Document changes to deprecate id SAPI solver resource field and replace with identity (#369)
dwavesystems/dwave-ocean-sdk: docs/shared/sapi_rest.rst

    solved_on       If this problem is in terminal state (``COMPLETED``,
                    ``CANCELLED`` or ``FAILED``), time when problem was solved
                    or cancelled.

>     solver          Unique identifier (name) of the solver. For resource representation
> 
>                     ``version=3``, the solver name is represented by the
>                     following JSON structure::
> 
>                     "solver": {"name": solver_name, "version": {"graph_id": graph_id}
> 
>                     where the fields are as follows:
>                    
>                     *    ``solver``: Parent for solver-related fields.       
>                     *    ``name``: Name of the solver.
>                     *    ``version``: Parent for QPU-version-related fields.
>                          This field does not apply to hybrid solvers.
>                     *   ``graph_id``: Identifier of the QPU solver's
>                         :ref:`working graph <topologies_working_graph>`.
> 
>                     For resource representation ``version=2`` and
>                     earlier, the solver name is represented as a string.
>                     The ``version=2`` resource representation is deprecated;
>                     instead, use the ``version=3`` JSON structure.

Reviewer: I pointed out this in the first review as well. We can't have an incomplete JSON value here. Check the [specs](https://www.json.org/json-en.html) for ref.
```suggestion
    solver          Unique identifier (name) of the solver. For resource representation
                    ``version=3``, the solver is represented by the
                    following JSON object::

                    {"name": name, "version": {"graph_id": graph_id}}

                    where the fields are as follows:
                   
                    *    ``name``: Name of the solver.
                    *    ``version``: Version object, used for QPU solvers only.
                    *   ``graph_id``: Identifier of the QPU solver's
                        :ref:`working graph <topologies_working_graph>`.

                    For resource representation ``version=2`` and
                    earlier, the solver is identified with a string.
                    The ``version=2`` resource representation is deprecated;
                    instead, use the ``version=3`` JSON structure.
```

    ============== ==============================
    avg_load       Average current load for the solver.
    description    Description of the solver.

>     id             Unique ID (name) of the solver. Only available for an
>                    ``Accept`` request header  with ``version=2.0.0``, which is
>                    the default.

Reviewer: Representation version varies per resource. Some are 2.0, some 2.1, etc, but they are all currently 2.x. In the future they'll be 3.x.

Also, we should refer to "resource representation version", not the accept header:

```suggestion
    id             Unique ID (name) of the solver. Only available in
                   solver representation versions 1 and 2.
```

    properties     :ref:`Solver properties <qpu_solver_properties_all>`
                   that reside in the ``properties`` dict; for example,
                   supported problem types, active qubits, active couplers,
                   total number of qubits, and so on.
    identity       Represents the following JSON structure. Only available for

>                    an ``Accept`` request header with ``version=3.0.0``.

Reviewer: Same comment as above.

                   .. code-block:: json
                    

>                         "identity":{
>                             "name":
>                             "version":{
>                                 "graph_id":
>                             }
>                         }

Reviewer: ```suggestion
                        "identity": {
                            "name": <name>,
                            "version": {
                                "graph_id": <graph-id>
                            }
                        }
```

                    
                   where the fields are as follows:
                   
                   *    ``identity``: Parent for solver-related fields.
                   
                   *    ``name``: Name of the solver.

>                    *    ``version``: Parent for version-related fields.

Reviewer: Version too is not relevant/present for hybrid solvers.

    ============== ==============================
    avg_load       Average current load for the solver.
    description    Description of the solver.
    id             Unique Identifier (name) of the solver, including its version.

                   Only available in resource representation ``version=2`` and
                   earlier. This field is deprecated; instead, use the
                   ``identity`` solver resource field.

>     identity       Represents the following JSON structure. Only available for
>                    resource representation ``version=3``.
> 
>                    .. code-block:: json
>                     
>                         "identity": {
>                             "name":
>                             "version": {
>                                 "graph_id": 
>                             }
>                         }
>                     
>                    where the fields are as follows:
>                    
>                    *    ``identity``: Parent for solver-related fields.
>                    
>                    *    ``name``: Name of the solver.
> 
>                    *    ``version``: Parent for QPU-version-related fields. This
>                         field does not apply to hybrid solvers.
>                    
>                    *    ``graph_id``: ID of the QPU solver's
>                         :ref:`working graph <topologies_working_graph>`.

Reviewer: ```suggestion
    identity       Represents the following JSON object. Only available for
                   resource representation ``version=3``.

                   .. code-block:: json
                    
                        {
                            "name": name
                            "version": {
                                "graph_id": graph_id
                            }
                        }
                    
                   where the values are as follows:
                   
                   *    ``name``: Name of the solver.

                   *    ``version``: Version object, used for QPU solvers only.
                   
                   *    ``graph_id``: ID of the QPU solver's
                        :ref:`working graph <topologies_working_graph>`.

```
